### ISSUE TITLE
Test cross compilation with `proc-macro` crates

### ISSUE THREAD

**Issue by jseyfried** [rust-lang/rust#37993]
c.f. #37958

**Comment by est31** [comment]
Can you actually somehow distinguish between target and host on proc-macro crates?

**Comment by jseyfried** [comment]
@est31 I think we shouldn't support cross compiling `proc-macro` crates -- that is, we always compile `proc-macro` crates for the host architecture. This issue is calling for testing of cross compiled crates that depend on `proc-macro` crates (which are compiled for the host) as well as other dependencies (which compiled for the target).

**Comment by emk** [comment]
I've posted a [stripped-down example program which triggers this error](https://github.com/rust-lang/rust/issues/37958#issuecomment-263063825).

**Comment by Mark-Simulacrum** [comment]
This seems like a E-needstest, probably sufficient to just have a run-make test I guess, which will then be tested on the cross test platforms. @jseyfried Does that sound correct? Please untag as needstest if that's wrong.

**Comment by jseyfried** [comment]
@Mark-Simulacrum Yeah, that sounds correct -- a run-make test is probably best here.

**Comment by steveklabnik** [comment]
Triage: not aware of anything here changing, but I'm also not an expert.
